src/operation/find/mod.rs

```diff
@@ -89,24 +89,29 @@ impl Operation for Find {
             body.insert("filter", filter.clone());
         }
 
-        Ok(Command::new(
+        Ok(Command::new_read(
             Self::NAME.to_string(),
             self.ns.db.clone(),
+            self.options
+                .as_ref()
```

Comment:
How much work would it be to separate out that work into a separate PR that could be merged and then rebased? If it's non-trivial, no need to do it, but if it's not too hard, it will make the diff for this PR a lot cleaner.

Reply:
Opened #169 containing both the write concern stuff and the read preference fix. For the write concern, I decided to just include the acknowledgment detection since that's really all the operation layer needs.

Reply:
Awesome, thanks